### PR TITLE
N-dims qgemm on x86 platform

### PR DESCRIPTION
**Pull request by chenbohua3**

### CODE REVIEW

File: tao_compiler/mlir/xla/ral/context/common_context_impl_quantization.cc [hunk 1]
```diff
@@ -1376,20 +1407,22 @@ MemRefType<int8_t, 2> ral_pdll_qgemm_onednn_s8_s8_s8_f32_per_channel(
...
     ExecutionContext* ctx, opaque_t /*stream_handle*/,
-    MemRefType<int8_t, 2> input, MemRefType<int8_t, 2> weight,
+    MemRefType<int8_t, NDims> input, MemRefType<int8_t, 2> weight,
     MemRefType<float, 0> inputScales, MemRefType<int32_t, 0> inputZeroPoints,
     MemRefType<float, 1> weightScales, MemRefType<int32_t, 1> weightZeroPoints,
     MemRefType<float, 0> resultScales, MemRefType<int32_t, 0> resultZeroPoints,
     void* customAttrs) {
   CpuTimer timer("ral_pdll_qgemm_onednn_s8_s8_s8_per_channel");
-  int64_t resultSizes[2] = {0, 0};
+  int64_t resultSizes[NDims];
```
Comment (wyzero) [inline tao_compiler/mlir/xla/ral/context/common_context_impl_quantization.cc:1419]: just use `int64_t resultSizes[NDims] = {0};` is ok.

**Review by wyzero — APPROVED** [review]

File: tao_compiler/mlir/xla/ral/context/common_context_impl_quantization.cc [hunk 2]
```diff
@@ -1314,7 +1327,13 @@ MemRefType<int8_t, 2> ral_pdll_qgemm_onednn_s8_s8_s8_f32_per_channel(
...
   auto driver = ctx->getDriver<cpu::CPUDriver>(cpu::CPUDriver::name());
   auto data = static_cast<int8_t*>(driver->alloc(ctx, m * n * sizeof(int8_t)));
-  auto result = assignMemRef<int8_t, 2>(data, resultSizes);
+  int64_t gemmResultSizes[2];
+  gemmResultSizes[0] = m;
+  gemmResultSizes[1] = n;
+  int64_t gemmInputSizes[2] = {m, k};
+  auto gemmResult = assignMemRef<int8_t, 2>(data, gemmResultSizes);
+
+  // auto result = assignMemRef<int8_t, 2>(data, resultSizes);
```
Comment (wyzero) [inline tao_compiler/mlir/xla/ral/context/common_context_impl_quantization.cc:1336]: nit : remove unused code.

File: tao_compiler/mlir/xla/ral/context/common_context_impl_quantization.cc (hunk 1 above)
Comment (chenbohua3) [reply tao_compiler/mlir/xla/ral/context/common_context_impl_quantization.cc]: done

File: tao_compiler/mlir/xla/ral/context/common_context_impl_quantization.cc (hunk 2 above)
Comment (chenbohua3) [reply tao_compiler/mlir/xla/ral/context/common_context_impl_quantization.cc]: done

**Review by wyzero — APPROVED** [review]

**Pull request merged**